rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -122,28 +98,46 @@ function _M.init_worker()
   end
 end
 
-function _M.call()
-  local phase = ngx.get_phase()
-  if phase == "log" then
-    after_balance()
-    return
+function _M.rewrite()
+  local balancer = get_balancer()
+  if not balancer then
+    ngx.status = ngx.HTTP_SERVICE_UNAVAILABLE
+    return ngx.exit(ngx.status)
```

Comment:
I'm not super familiar with the ingress-nginx config, but I'm guessing this means that _every_ configured `upstream` will be configured to use Lua load balancing? Likewise, will `rewrite_by_lua` be called for every configured site? Why not move this logic into `balance` instead?

Reply:
@csfrancis this was in `balance()` before. But it turns out modifying `ngx.status` is not allowed in `balance` phase.

> but I'm guessing this means that every configured upstream will be configured to use Lua load balancing

that's correct it's either all or none